### DISCUSSION TITLE
Log output query

### DISCUSSION THREAD

**Discussion by kaihendry** [KyleRoss/node-lambda-log#58] [Q&A] [answered]
```
2021-09-21T01:30:02.820Z	undefined	WARN	{
    "level": "warn",
    "msg": "a warning!",
    "timestamp": "2021-09-21T01:30:02.820Z",
    "_tags": []
}
```

Why is there these extra fields like 
* 2021-09-21T01:30:02.820Z - redundant since my logs already have a timestamp by AWS
* "undefined" - I'm really not sure what the purpose of this is
* WARN - this is already in my structure log, why is it here? make it easier to filter?

**Comment by KyleRoss** [comment, marked as answer]
- _2021-09-21T01:30:02.820Z - redundant since my logs already have a timestamp by AWS_
This timestamp field is not added by LambdaLog by default, it appears you have set metadata somewhere in your application that has added this field to the log message.
- _"undefined" - I'm really not sure what the purpose of this is_
I'm not sure where the `undefined` is coming from. I've been using this project in several enterprise applications and haven't encountered this issue. Given that it's shown before the `WARN` in your example, I'd say it's something with CloudWatch as we cannot change anything before the `WARN`.
- _WARN - this is already in my structure log, why is it here? make it easier to filter?_
Since AWS logs in CloudWatch are labeled by the `console` method used, not all log levels within LambdaLog (including custom ones) would be labeled properly in CloudWatch because there is not a corresponding `console` method. For example, if you have a custom log level of `fatal`, we couldn't use `console.fatal()` to log the message as it does not exist, in this case, we would use the next best thing, `console.error()`. This would label the log message with `ERROR` instead of `fatal` in AWS which wouldn't allow you to filter by `fatal` messages.
